src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1660,7 +1740,19 @@ def generate(
         if generation_config.return_timestamps:
             logits_processor = [WhisperTimeStampLogitsProcessor(generation_config)]
 
-        return super().generate(
+        if return_token_timestamps:
+            kwargs["output_attentions"] = True
+            kwargs["return_dict_in_generate"] = True
+
+        if return_token_timestamps:
+            if getattr(generation_config, "task", None) == "translate":
+                logger.warning("Token-level timestamps may not be reliable for task 'translate'.")
+            if not hasattr(self.config, "alignment_heads"):
+                logger.warning(
+                    "Model configuration has no `alignment_heads`, token-level timestamps not available. See https://gist.github.com/hollance/42e32852f24243b748ae6bc1f985b13a on how to add this property to the model config."
```

Comment:
(nit) Can we split this warning over two lines!
```suggestion
                    "Model configuration has no `alignment_heads`, token-level timestamps not available."
                    " See https://gist.github.com/hollance/42e32852f24243b748ae6bc1f985b13a on how to add this property to the model config."
```